### PR TITLE
[python] Add user-level stats hooks, to reach parity with tiledbsoma-r

### PR DESCRIPTION
**Pull request by johnkerl**
## Issue and/or context:

Issue: #514 

## Changes:

Bring this one step closer to the user, following the R lead at #865

### CODE REVIEW

File: apis/python/src/tiledbsoma/__init__.py [hunk 1]
```diff
@@ -14,6 +14,12 @@
 )
 from .measurement import Measurement
 from .sparse_nd_array import SparseNDArray
+from .util_tiledb import (
+    stats_disable,
+    stats_dump,
+    stats_enable,
+    stats_reset,
+)
```
Comment (thetorpedodog) [inline apis/python/src/tiledbsoma/__init__.py:22]: Would it make sense to export these directly from `libtiledbsoma` and skip the indirection in `util_tiledb.py`?

```suggestion
from .libtiledbsoma import (
    stats_disable,
    stats_dump,
    stats_enable,
    stats_reset,
)
```

Docstrings [can be created directly at the libtiledbsoma level](https://pybind11.readthedocs.io/en/stable/advanced/misc.html#generating-documentation-using-sphinx).

Comment (johnkerl) [reply apis/python/src/tiledbsoma/__init__.py:22]: super-snazz re-export @thetorpedodog 😎

Comment (johnkerl) [reply apis/python/src/tiledbsoma/__init__.py:22]: ... or nah 😐 

```
Help on built-in function stats_enable in module tiledbsoma.libtiledbsoma:

stats_enable(...) method of builtins.PyCapsule instance
    stats_enable() -> None
```

Doing the re-export leaves us without docstrings, b/c from `pybind11`

Doing it the naïve way I did (wrap & delegate) does give us docstrings (which I forgot to replace from XXX which is my way of saying to-do to myself, sorry for posting this PR prematurely)

Comment (thetorpedodog) [reply apis/python/src/tiledbsoma/__init__.py:22]: as I said, [docstrings can be created directly in pybind11](https://pybind11.readthedocs.io/en/stable/advanced/misc.html#generating-documentation-using-sphinx), saving a layer of copypasting.

Comment (johnkerl) [reply apis/python/src/tiledbsoma/__init__.py:22]: My apologies @thetorpedodog , missed that line -- looking now ...

**Comment by johnkerl** [comment]
@thetorpedodog round three 🙏

File: apis/python/src/tiledbsoma/util_tiledb.py [hunk 2]
```diff
@@ -8,6 +8,7 @@
 from pandas.api.types import infer_dtype, is_categorical_dtype
 from somacore import options
 
+# This package's pybind11 code
```
Comment (thetorpedodog) [inline apis/python/src/tiledbsoma/util_tiledb.py:11]: outdated comment

**Review by thetorpedodog — APPROVED** [review]

**Review by eddelbuettel — APPROVED** [review]
LGTM with the usual caveat that I am not the one you ask for _idiomatic_ python advice...

**Comment by johnkerl** [comment]
@eddelbuettel idiomatic python advice is only one kind of advice ... i appreciate your insight always, especially in an r/python parity context :)

**Pull request merged**